sklearn/cluster/hierarchical.py

```diff
@@ -756,6 +765,14 @@ def __init__(self, n_clusters=2, affinity="euclidean",
         self.affinity = affinity
         self.pooling_func = pooling_func
 
+    @property
+    @deprecated("The `n_components` parameter changed to "
```

Comment:
Please use the following as the message:

```
The ``n_components_`` attribute was deprecated in in favor of ``n_connected_components_`` in 0.21 and will be removed in 0.23.
```

- you need double backticks to avoid linking to the glossary
- some very strange bug cuts the docstring after the first period. I'll open an issue for this.

Reply:
regarding the bug I opened #13489

Reply:
Got it Nicolas, I changed this in commit [#bdbb14a](https://github.com/scikit-learn/scikit-learn/pull/13427/commits/bdbb14a28f9f9a7db9d2893b28bb43caa65aceba) and changed the string to match the change in the test in commit [#993fc35](https://github.com/scikit-learn/scikit-learn/pull/13427/commits/993fc3530536a7289d6e8e5fdd02a98e808200bf)